### ISSUE TITLE
False positives reported for subscripts in maths

### ISSUE THREAD

**Issue by cfr42** [Witiko/expltools#18]
Minimised (so evidently silly) example:
```latex
\documentclass{article}
\ExplSyntaxOn
\tl_new:N \l_test_tmpa_tl
\ExplSyntaxOff
\begin{document}
$ \sum_n $
$ \sum_{n}$
\[
\sum_{n}
\]
\[
\sum_n
\]
\end{document}

% vim: et:sw=2:ts=2:foldmethod=marker:
```
For this file `explcheck` reports
```bash
Checking 1 file

Checking explcheck-test.tex                                     2 errors

    explcheck-test.tex:6:3:              E102 expl3 control sequences in non-expl3 parts
    explcheck-test.tex:12:1:             E102 expl3 control sequences in non-expl3 parts

Total: 2 errors, 0 warnings in 1 file
```
The reports of errors are picking up genuinely incorrect syntax in the file. `\sum_n` is not officially supported. However, 
1. such syntax is common and does not meet `explcheck`'s criterion for an error: according to the documentation, an error is likely to result in a compilation error;
2.  the mistake is certainly not the error reported by `explcheck`: `\sum_n` may be bad syntax, but it isn't an `expl3` control sequence;
3. `\sum_n` does not even look like an `expl3` control sequence, which would need at least a colon to qualify.

If I add `\l_test_tmpa_tl` before `\end{document}`, `explcheck` reports three E102 errors, but `\l_test_tmpa_tl` isn't an `expl3` control sequence either, though it is definitely an error in a non-`expl3` part. 

Should `E102` say `expl3 control sequences or variables`?

**Comment by Witiko** [comment]
Thank you for the report. This is definitely an error, the detection of expl3 control sequences should be definitely more precise. At the very least, we should only control sequences that look like functions (`\<module>_<description>:<argspec>`) or variables/constants (`\<scope>_<module>_<description>_<type>`). If we wanted to be extra careful, we might only report predefined control sequences of l3kernel.

> `\l_test_tmpa_tl` isn't an `expl3` control sequence either, though it is definitely an error in a non-`expl3` part.
> 
> Should `E102` say `expl3 control sequences or variables`?

[A control sequence][1] refers to essentially anything that starts with a backslash (`\`). It covers expl3 functions, variables, constants, quarks, etc.

 [1]: https://en.wikibooks.org/wiki/TeX/definition/control_sequence

**Comment by cfr42** [comment]
> If we wanted to be extra careful, we might only report predefined control sequences of l3kernel.

I don't think this would be especially helpful, unless the linter is only really targeted at kernel code. For functions, at least, I'd assume you could use the colon to make the test a bit more precise. For something like a variable or constant or whatever, it's a bit trickier. But you could at least rule out things which don't start with one of the standard prefixes.

Or maybe ignore maths mode in non-expl3 parts?

> [A control sequence](https://en.wikibooks.org/wiki/TeX/definition/control_sequence) refers to essentially anything that starts with a backslash (`\`). It covers expl3 functions, variables, constants, quarks, etc.

I wondered about that, but `expl3` terminology is usually more precise, so I was expecting it to report functions. But that's evidently a misconception on my part.

**Comment by Witiko** [comment]
> I wondered about that, but `expl3` terminology is usually more precise, so I was expecting it to report functions. But that's evidently a misconception on my part.

I am sorry to confuse. Writing "control sequence" felt more economical than typing out all the expl3 concepts (functions, variables, constants, ...) that are written as TeX control sequences in expl3.

**Comment by Witiko** [comment]
> What about 'expl3 syntax'?

I am sorry, I already renamed it to "expl3 material" in commit https://github.com/Witiko/expltools/commit/26db76b7b21339432b50904782e111fd121f78ed 3 hours prior to your comment in explcheck 0.3.0, which has since been released on CTAN. Regardless, I am slightly in favor of the name "material", as it implies valid expl3 content, whereas expl3 syntax can refer to anything that resembles expl3 material. While the latter may be more accurate vis-à-vis the current implementation, the aspiration is to have a precise classifier that only matches actual expl3 material/content.
